### PR TITLE
CAS Munitions Revamp, Addition of Bombs.

### PR DESCRIPTION
**Pull request by Pariah919**
## About The Pull Request
Quick overview of changes so far.
- HV is 2 second travel time now.

- Missiles have been split into classes.
- High Speed Missiles which are called AGMs, they have a typical travel time of 3 seconds, 2/3/4 as the standard. Keeper is the bog standard for this. Widowmaker is the specialized murder rocket at 2 second travel time with less outer ring, and Napalm does fire things. Their lase style is a dot ->dot in rapid sequence.
- High Yield Low speed, which are called PGHMs, the (new!) Monarch missile falls into here being the standard 3/5/7 and 6 second travel time. The other ones here are the Fatty and Banshee, which both have been changed to fit the PGHM class. Their dot style is the old quad dot.
- Normal Missiles, the only one here is the (New!) Swansong, which is 2/4/6 with 4 second travel time. This is the bogstandard everyday missile category, which is funny since it has only one for now. Their dot style is a ^

I also added bombs and bomblets. They have their own pylons. I added a bomblet/bomb thrower to every shipside map.
- Bomb/lets are designed around their high travel times of 10 to 12 seconds.
- Bomblets range from 50 to 199lb. 0.5 fire delay between bomblet. Their dot style is a rapid |.
- 50lb Bombs are 2 light, carry 40 bomblets. 10 second travel time
- 75lb are 3 light, carry 20 bomblets. 12 second.

- Bombs range from 200 to 1000lb. Their dot shape is a normal dot rotating in place while spazzing out in and out of existance.
- 200lbs are the bog standard bomb, 12 second travel time. 0/3/4. 8 per pylon.
- 400 bombs are the above average in payload bomb, 12 second travel time. 0/4/5. 4 per pylon.
- 1000lb bombs are the mother of all bombs. 6/9/0. 2 per pylon. Has its very own lase! If you see a red circle. Run. Very far. 14 second travel time.

As with anything these values are possibly subject to change

## Why It's Good For The Game
This was originally going to be a mere addition of bombs, but then I saw how horrifically unstandardized CAS is.
The issues to be frank with old CAS is that nothing is stratified into classes alongside being dependent on orbit balance, but at the very core is the fact that we've basically been dragging along whatever random values CM assigned like, years ago. 

My aim is to clearly stratify every missile type into clear classes, and dether all CAS weapons from the dregs of orbit balance entirely.
Bombs are cool, as a sidenote. Have fun. Go blow up the marine force with MOAB bombs or something.

## Changelog
:cl:
add: Two new type of missiles, the Swansong and Monarch.
add: CAS bombs and bomblets. 200, 400, and 1000lb bombs. 50lb and 75lb bombs.
balance: CAS Missiles have been entirely rebalanced into classes.
balance: The classes are ATGM (3s/AGM), Heavy(6s/PGHM), Normal (4s/PGLM).
/:cl:

### CODE REVIEW

**Comment by dopamiin0** [comment]
as the ceo of cas i am here to approve of this pr and support the disastrous consequences it will bring

**Comment by FosterGrime-Heymynameslouie** [comment]
bomba

**Comment by Oscari-pog** [comment]
Missing 4000lb bomb

**Comment by Pariah919** [comment]
> Missing 4000lb bomb

Nah that's the 4000lb PGHM, soon :TM:

**Comment by yyzsong** [comment]
Good but remove the CAS warning marines get. Then it'll be perfect.

**Comment by isotope436** [comment]
> Good but remove the CAS warning marines get. Then it'll be perfect.

well this is a joke i fully support the removal of the warning system for xenos and marines

**Comment by SimplySmiley123** [comment]
> Good but remove the CAS warning marines get. Then it'll be perfect.

Remove it from benos too for maximum funny 👍

**Comment by yyzsong** [comment]
> > Good but remove the CAS warning marines get. Then it'll be perfect.
> 
> Remove it from benos too for maximum funny 👍

Marines get a different kind of CAS warning that just tells them the area of the explosion. Xenos don't. That's what I'm talking about removing.

File: code/game/objects/structures/dropship_ammo.dm [hunk 1]
```diff
@@ -612,3 +640,97 @@
 
 /obj/structure/ship_ammo/cas/minirocket/illumination/proc/drop_cas_flare(turf/impact)
 	new /obj/effect/temp_visual/above_flare(impact)
+
+// Bombs have a long travel time but are decently numerous, ranging in payloads from 200 to 1000lbs. Higher is bigger.
+/obj/structure/ship_ammo/cas/bomb
+	name = "AOE-200lb 'Tiny' stack"
+	desc = "A decent-sized payload of explosive bombs, will only fit in a full-sized bomb pod. Moving this will require some sort of lifter."
```
Comment (TiviPlus) [inline code/game/objects/structures/dropship_ammo.dm:647]: lot of these are missing \improper

File: code/game/objects/structures/dropship_equipment.dm [hunk 2]
```diff
@@ -867,3 +867,43 @@
...
+	point_cost = 450
+	dropship_equipment_flags = USES_AMMO|IS_WEAPON|IS_INTERACTABLE
+	ammo_type_used = CAS_BOMBLET
+
+/obj/structure/dropship_equipment/cas/weapon/bomblet_pod/update_icon()
+	if(ammo_equipped?.ammo_count)
+		icon_state = "bomblet_pod_loaded"
+	else
+		if(ship_base)
+			icon_state = "bomblet_pod_installed"
```
Comment (TiviPlus) [inline code/game/objects/structures/dropship_equipment.dm:887]: use elseif

File: code/game/objects/structures/dropship_equipment.dm [hunk 3]
```diff
@@ -867,3 +867,43 @@
...
+	firing_sound = 'sound/weapons/gunship_rocketpod.ogg'
+	firing_delay = 2 SECONDS
+	point_cost = 450
+	dropship_equipment_flags = USES_AMMO|IS_WEAPON|IS_INTERACTABLE
+	ammo_type_used = CAS_BOMB
+
+/obj/structure/dropship_equipment/cas/weapon/bomb_pod/update_icon()
+	if(ammo_equipped?.ammo_count)
+		icon_state = "bomb_pod_loaded"
+	else
```
Comment (TiviPlus) [inline code/game/objects/structures/dropship_equipment.dm:905]: ditto

**Review by TiviPlus — CHANGES_REQUESTED** [review]
rest lgtm

**Comment by TiviPlus** [comment]
also get kuro to review

**Review by Kuro020 — APPROVED** [review]

File: code/game/objects/structures/dropship_equipment.dm [hunk 4]
```diff
@@ -867,3 +867,41 @@
...
+	icon_state = "bomb_pod"
+	desc = "A bomb pod capable of launching several large bombs. Moving this will require some sort of lifter."
+	icon = 'icons/Marine/mainship_props64.dmi'
+	firing_sound = 'sound/weapons/gunship_rocketpod.ogg'
+	firing_delay = 2 SECONDS
+	point_cost = 450
+	dropship_equipment_flags = USES_AMMO|IS_WEAPON|IS_INTERACTABLE
+	ammo_type_used = CAS_BOMB
+
+/obj/structure/dropship_equipment/cas/weapon/bomb_pod/update_icon()
```
Comment (ivanmixo) [inline code/game/objects/structures/dropship_equipment.dm:901]:
```suggestion
/obj/structure/dropship_equipment/cas/weapon/bomb_pod/update_icon_state()
```
i literally told you this in dms idk why you ignored it

File: code/game/objects/structures/dropship_equipment.dm [hunk 5]
```diff
@@ -867,3 +867,41 @@
...
+	icon_state = "bomblet_pod"
+	desc = "A pnuematic thrower machine capable of up to 40 smaller bombs, generally  called 'bomblets'. Moving this will require some sort of lifter."
+	icon = 'icons/Marine/mainship_props64.dmi'
+	firing_sound = 'sound/weapons/gunship_rocketpod.ogg'
+	firing_delay = 0.5 SECONDS
+	point_cost = 450
+	dropship_equipment_flags = USES_AMMO|IS_WEAPON|IS_INTERACTABLE
+	ammo_type_used = CAS_BOMBLET
+
+/obj/structure/dropship_equipment/cas/weapon/bomblet_pod/update_icon()
```
Comment (ivanmixo) [inline code/game/objects/structures/dropship_equipment.dm:882]: same here

**Review by ivanmixo — CHANGES_REQUESTED** [review]
idk how i feel about such low travel times but we'll see ig

**Comment by Arturlang** [comment]
I think the widowmaker, or whichever dot was grey, really needs a better sprite that isn't as hidden

File: code/game/objects/structures/dropship_equipment.dm (hunk 4 above)
Comment (Pariah919) [reply code/game/objects/structures/dropship_equipment.dm:901]: i had already pushed it you goober

**Review by TiviPlus — APPROVED** [review]

**Comment by chizzy376** [comment]
highfleet gaming, can't wait to see more CAS variety

**Comment by OpsCoretango** [comment]
You a sigma, bro

**Pull request merged**